### PR TITLE
Payara InSight Enterprise UI/UX

### PR DESCRIPTION
**Pull request by jbee**
### Summary
Complete UI/UX overhaul of Mon.. InSight. 

What can I say: most things UI/UX did change. Except the graphs themselves stayed pretty much what they were.

On the Java/REST API side of things there is only a tiny change adding a new field to the watch state making it possible to show how long the watch has been in its current state.

All other changes are client side and only affect the rendering and behaviour of the application in the browser.

A summary:

* page got a copyright footer
* page got a favicon
* app includes and uses custom font
* app includes and uses SVG icons and new logo SVGs
* navigation and quick settings in left sidebar
* settings in right sidebar now utilise tabs and got reorganised as with tabs you only can look at a single group at a time
* buttons to add new pages, widgets, watches are placed and marked more prominently
* increased use of modal dialogs
* destructive actions use a confirmation dialog (shown red if potentially causes a server state change)
* modifying actions show success or error messages
* some settings checkboxes are now rendered as toggle switches (where it made sense)
* query based pages can now use a widget type filter (was needed to produce a "clean" RAG page)
* RAG page is now called _Traffic Light Status_ as _RAG_ wasn't the common term I took it to be
* ... lots of styling details

### Testing
1. build this PR
2. in order to use InSight from this PR change the root `pom.xml` of Payara Enterprise accordingly:
```xml
-        <monitoring-console-api.version>1.1</monitoring-console-api.version>
-        <monitoring-console-process.version>1.2.1</monitoring-console-process.version>
-        <monitoring-console-webapp.version>1.2.1</monitoring-console-webapp.version>
+        <monitoring-console-api.version>1.2-SNAPSHOT</monitoring-console-api.version>
+        <monitoring-console-process.version>1.4-SNAPSHOT</monitoring-console-process.version>
+        <monitoring-console-webapp.version>1.4-SNAPSHOT</monitoring-console-webapp.version>
```
3. build and deploy Payara Enterprise with changed pom
4. start it using `asadmin start-domain production`
5. open admin-console, click link _Open Monitoring Console_ on the dashboard

Now you should see the new InSight (might take a browser refresh if you have been testing MC before).
Also you might want to delete local storage for the page to get rid of any old state.

From here just go crazy and try to click on everything ;) 
It might be a good idea to do so as _User_ and later also test as _Administrator_ (Role switched in app settings).

### CODE REVIEW

File: webapp/src/main/webapp/js/mc-view.js [hunk 1]
```diff
@@ -341,31 +359,42 @@ MonitoringConsole.View = (function() {
...
+            
         ]});
         settings.push({ id: 'settings-status', caption: 'Status', collapsed: true, description: 'Set a text for an assessment status', entries: [
-            { label: '"No Data"', type: 'text', value: widget.status.missing.hint, onChange: (widget, text) => widget.status.missing.hint = text},
-            { label: '"Alaraming"', type: 'text', value: widget.status.alarming.hint, onChange: (widget, text) => widget.status.alarming.hint = text},
-            { label: '"Critical"', type: 'text', value: widget.status.critical.hint, onChange: (widget, text) => widget.status.critical.hint = text},
+            { label: '"No Data"', type: 'textarea', value: widget.status.missing.hint, onChange: (widget, text) => widget.status.missing.hint = text},
+            { label: '"Alaraming"', type: 'textarea', value: widget.status.alarming.hint, onChange: (widget, text) => widget.status.alarming.hint = text},
```
Comment (Pandrex247) [inline webapp/src/main/webapp/js/mc-view.js]:
```suggestion
            { label: '"Alarming"', type: 'textarea', value: widget.status.alarming.hint, onChange: (widget, text) => widget.status.alarming.hint = text},
```
Typo

File: webapp/src/main/webapp/js/mc-view.js [hunk 2]
```diff
@@ -160,93 +142,95 @@ MonitoringConsole.View = (function() {
...
-            { label: 'Page Sync', available: pushAvailable || pullAvailable, input: [
-                { available: pushAvailable, input: () => $('<button />', { text: 'Update Remote Pages', title: 'Push local state of all know remote pages to server'}).click(MonitoringConsole.Model.Page.Sync.pushAllLocal) },
-                { available: pullAvailable, input: () => $('<button/>', { text: 'Update Local Pages', title: 'Open Page synchronisation dialoge'}).click(onPagesSync) }, 
+            { id: 'settings-pages', available: pushAvailable || pullAvailable, type: 'app', caption: 'Pages', entries: [
+                { label: 'Server Sync', input: [
+                    { available: pushAvailable, input: () => $('<button />', { text: 'Update Server...', title: 'Push local state of all know remote pages to server'}).click(showPagePushModalDialog) },
+                    { available: pullAvailable, input: () => $('<button/>', { text: 'Update Local...', title: 'Open Page synchronisation dialoge'}).click(showPageSyncModalDialog) },
```
Comment (Pandrex247) [inline webapp/src/main/webapp/js/mc-view.js]:
```suggestion
                    { available: pullAvailable, input: () => $('<button/>', { text: 'Update Local...', title: 'Open Page synchronisation dialogue'}).click(showPageSyncModalDialog) }, 
```
Typo

Comment (Pandrex247) [reply webapp/src/main/webapp/js/mc-view.js]: Separately, might want to think about the capitalisation
Is Page a noun in this instance?
Should we be using Title Case?

File: webapp/src/main/webapp/js/mc-data.js [hunk 3]
```diff
@@ -66,16 +66,16 @@ MonitoringConsole.Data = (function() {
     /**
      * Description texts used in the preset pages...
      */
-	const TEXT_HTTP_HIGH = "Requires *HTTP monitoring* to be enabled: Goto _Configurations_ => _Monitoring_ and set *'HTTP Service'* to *'HIGH'*.";
-	const TEXT_WEB_HIGH = "Requires *WEB monitoring* to be enabled: Goto _Configurations_ => _Monitoring_ and set *'Web Container'* to *'HIGH'*.";
+	const TEXT_HTTP_HIGH = "Requires *HTTP monitoring* to be enabled: Goto _Configurations_ => _Monitoring_ and set *HTTP Service* to *HIGH*.";
+	const TEXT_WEB_HIGH = "Requires *WEB monitoring* to be enabled: Goto _Configurations_ => _Monitoring_ and set *Web Container* to *HIGH*.";
 	const TEXT_REQUEST_TRACING = "If you did enable request tracing at _Configurations_ => _Request Tracing_ not seeing any data means no requests passed the tracing threshold which is a good thing.";
```
Comment (Pandrex247) [inline webapp/src/main/webapp/js/mc-data.js]:
```suggestion
	const TEXT_REQUEST_TRACING = "If you enabled request tracing at _Configurations_ => _Request Tracing_, not seeing any data means no requests passed the tracing threshold which is a good thing.";
```

File: webapp/src/main/webapp/js/mc-data.js [hunk 4]
```diff
@@ -66,16 +66,16 @@ MonitoringConsole.Data = (function() {
...
+	const TEXT_CPU_USAGE = "Requires *CPU Usage HealthCheck* to be enabled: Goto _Configurations_ => _HealthCheck_ => _CPU Usage_ tab and check *enabled*";
+	const TEXT_HEAP_USAGE = "Requires *Heap Usage HealthCheck* to be enabled: Goto _Configurations_ => _HealthCheck_ => _Heap Usage_ tab and check *enabled*";
+	const TEXT_GC_PERCENTAGE = "Requires *Garbage Collector HealthCheck* to be enabled: Goto _Configurations_ => _HealthCheck_ => _Garbage Collector_ tab and check *enabled*";
+	const TEXT_MEM_USAGE = "Requires *Machine Memory HealthCheck* to be enabled: Goto _Configurations_ => _HealthCheck_ => _Machine Memory Usage_ tab and check *enabled*";
+	const TEXT_POOL_USAGE = "Requires *Connection Pool HealthCheck* to be enabled: Goto _Configurations_ => _HealthCheck_ => _Connection Pool_ tab and check *enabled*";
+	const TEXT_LIVELINESS = "Requires *MicroProfile HealthCheck Checker* to be enabled: Goto _Configurations_ => _HealthCheck_ => _MicroProfile HealthCheck Checker_ tab and check *enabled*";
```
Comment (Pandrex247) [inline webapp/src/main/webapp/js/mc-data.js]:
```suggestion
	const TEXT_CPU_USAGE = "Requires *CPU Usage HealthCheck* to be enabled: Go to _Configurations_ => _HealthCheck_ => _CPU Usage_ tab and check *enabled*";
	const TEXT_HEAP_USAGE = "Requires *Heap Usage HealthCheck* to be enabled: Go to _Configurations_ => _HealthCheck_ => _Heap Usage_ tab and check *enabled*";
	const TEXT_GC_PERCENTAGE = "Requires *Garbage Collector HealthCheck* to be enabled: Go to _Configurations_ => _HealthCheck_ => _Garbage Collector_ tab and check *enabled*";
	const TEXT_MEM_USAGE = "Requires *Machine Memory HealthCheck* to be enabled: Go to _Configurations_ => _HealthCheck_ => _Machine Memory Usage_ tab and check *enabled*";
	const TEXT_POOL_USAGE = "Requires *Connection Pool HealthCheck* to be enabled: Go to _Configurations_ => _HealthCheck_ => _Connection Pool_ tab and check *enabled*";
	const TEXT_LIVELINESS = "Requires *MicroProfile HealthCheck Checker* to be enabled: Go to _Configurations_ => _HealthCheck_ => _MicroProfile HealthCheck Checker_ tab and check *enabled*";
```
Goto isn't actually a proper English word / abbreviation, 'tis a character in Star Wars though

File: webapp/src/main/webapp/js/mc-view.js [hunk 5]
```diff
@@ -388,82 +417,254 @@ MonitoringConsole.View = (function() {
...
+                { input: $('<button/>', {text: 'Update Now'}).click(() => configure(page => page.content.expires = undefined)) },
             ]},
-            { label: 'Add Widgets', available: !queryAvailable, input: addWidgetsInput },
-            { label: 'Sync', available: pushAvailable || pullAvailable, input: [
-                { available: autoAvailable, label: 'auto', type: 'checkbox', value: MonitoringConsole.Model.Page.Sync.auto(), onChange: (checked) => MonitoringConsole.Model.Page.Sync.auto(checked),
+            { label: 'Filter Type', available: queryAvailable, type: 'dropdown', options: WIDGET_TYPE_FILTER_OPTIONS, value: page.content.filter, onChange: filter => onPageUpdate(configure(page => page.content.filter = filter)) },
+            { label: 'Server Sync', available: pushAvailable || pullAvailable, input: [
+                { available: autoAvailable, label: 'auto', type: 'checkbox', value: Page.Sync.auto(), onChange: (checked) => Page.Sync.auto(checked),
                     description: 'When checked changed to the page are automatically pushed to the remote server (shared with others)' },
```
Comment (Pandrex247) [inline webapp/src/main/webapp/js/mc-view.js]:
```suggestion
                    description: 'When checked changes to the page are automatically pushed to the remote server (shared with others)' },
```
Grammar

File: webapp/src/main/webapp/js/mc-view.js [hunk 6]
```diff
@@ -388,82 +417,254 @@ MonitoringConsole.View = (function() {
...
-                { input: $('<button/>', {text: 'Update'}).click(() => configure(page => page.content.expires = undefined)) },
+                { input: $('<button/>', {text: 'Update Now'}).click(() => configure(page => page.content.expires = undefined)) },
             ]},
-            { label: 'Add Widgets', available: !queryAvailable, input: addWidgetsInput },
-            { label: 'Sync', available: pushAvailable || pullAvailable, input: [
-                { available: autoAvailable, label: 'auto', type: 'checkbox', value: MonitoringConsole.Model.Page.Sync.auto(), onChange: (checked) => MonitoringConsole.Model.Page.Sync.auto(checked),
+            { label: 'Filter Type', available: queryAvailable, type: 'dropdown', options: WIDGET_TYPE_FILTER_OPTIONS, value: page.content.filter, onChange: filter => onPageUpdate(configure(page => page.content.filter = filter)) },
+            { label: 'Server Sync', available: pushAvailable || pullAvailable, input: [
+                { available: autoAvailable, label: 'auto', type: 'checkbox', value: Page.Sync.auto(), onChange: (checked) => Page.Sync.auto(checked),
```
Comment (Pandrex247) [inline webapp/src/main/webapp/js/mc-view.js]:
```suggestion
                { available: autoAvailable, label: 'Auto', type: 'checkbox', value: Page.Sync.auto(), onChange: (checked) => Page.Sync.auto(checked),
```
Capitalisation

**Review by Pandrex247 — COMMENTED** [review]
Separate typos found which I can't comment on in this PR:
* ... more user ~~freindly~~ friendly ...
* ~~pause~~ Pause data updates
* ~~start~~ Start page rotation

And as noted in Teams, there is some odd behaviour surrounded the reset functionality with the Traffic Light Status page.

File: webapp/src/main/webapp/js/mc-view.js (hunk 2 above)
Comment (jbee) [reply webapp/src/main/webapp/js/mc-view.js]: Fixed this one manually as I also change `Page` to `page`.

**Comment by jbee** [comment]
@Pandrex247 typos are fixed. 

For the record: Reset issue also solved.

**Review by Pandrex247 — APPROVED** [review]
🎉

**Pull request merged**